public void testExecute() throws Exception {
    User user = new User();
    user.setUsername("self-registered");
    user.setPassword("Password1");
    user.setConfirmPassword("Password1");
    user.setFirstName("First");
    user.setLastName("Last");
    Address address = new Address();
    address.setCity("Denver");
    address.setProvince("CO");
    address.setCountry("USA");
    address.setPostalCode("80210");
    user.setAddress(address);
    user.setEmail("self-registered@raibledesigns.com");
    user.setWebsite("http://raibledesigns.com");
    user.setPasswordHint("Password is one with you.");
    action.setUser(user);
    // set mock response so setting cookies doesn't fail
    ServletActionContext.setResponse(new MockHttpServletResponse());
    // start SMTP Server
    Wiser wiser = new Wiser();
    wiser.setPort(getSmtpPort());
    wiser.start();
    assertEquals("success", action.save());
    assertFalse(action.hasActionErrors());
    // verify an account information e-mail was sent
    wiser.stop();
    assertTrue(wiser.getMessages().size() == 1);
    // verify that success messages are in the session
    assertNotNull(action.getSession().getAttribute(Constants.REGISTERED));
    SecurityContextHolder.getContext().setAuthentication(null);
}
